### PR TITLE
tests.check_vilain.c: Write tests for 'io_text'

### PR DESCRIPTION
**Pull request by DavidVn64**
I haven't split the tests with cases since tests are very basics and only depends on if a path of a file is a good one or not.

### CODE REVIEW

File: tests/check_vilain.c [hunk 1]
```diff
@@ -354,18 +354,85 @@ END_TEST Suite *buffer_suite(void)
...
+    ck_assert_ptr_eq(load_file(wrong_path2), NULL);
+    ck_assert_ptr_eq(load_file(wrong_path3), NULL);
+}
+
+END_TEST START_TEST(test_save_file)
+{
+    char *good_path = "text.txt";
+    char *wrong_path1 = "text";
+    char *wrong_path2 = "//";
+    char *wrong_path3 = "text.txt/text";
```
Comment (Kagamihime) [inline tests/check_vilain.c]: To declare a string in C you have to do
```c
char str[] = "/* your string */";
```

File: tests/check_vilain.c [hunk 2]
```diff
@@ -354,18 +354,85 @@ END_TEST Suite *buffer_suite(void)
     return s;
 }
 
+START_TEST(test_load_file)
+{
+    char *good_path = "text.txt";
```
Comment (Kagamihime) [inline tests/check_vilain.c]: You are testing the functon `load_file` but you didn't add any actual file to load.. It's not an actual test in fact...

File: tests/check_vilain.c (hunk 1 above)
Comment (Kagamihime) [inline tests/check_vilain.c]: There is not point at using "good" and "wrong" paths for your tests. What you have to test is if when you load a file, the content of the file is the same as the content of the buffer you got from `load_file`. And if the content of the buffer you saved with `save_file` is really in the file you wrote in

File: tests/check_vilain.c [hunk 3]
```diff
@@ -354,18 +354,85 @@ END_TEST Suite *buffer_suite(void)
...
+    ck_assert_int_eq(save_buffer(wrong_path3, buff), -1);
+
+}
+
+END_TEST Suite *io_text_suite(void)
+{
+    TCase *tc_io_text;
+    Suite *s;
+
+    tc_buffer = tcase_create("Core");
```
Comment (Kagamihime) [inline tests/check_vilain.c]: `tc_buffer` doesn't exist! And it should be named `tc_io_text` like in the rest of your funtion ^^

**Comment by Kagamihime** [comment]
You didn't run coala

**Comment by DavidVn64** [comment]
Oh yes my bad, but it is not finished, I forgot some cases and I didn’t implemented the function that calls the tests yet, it will be done today :)

**Comment by GaelleMarais** [comment]
Where are you with this issue ?

**Comment by DavidVn64** [comment]
I compiled and runned coala and Travis fails ?

**Comment by Kagamihime** [comment]
Yes, because your tests are failing. Two cases: your tests and/or the module io_text are not working correctly

**Comment by GaelleMarais** [comment]
This is the error message : `The command "make && cd tests && make test" exited with 2. `
Go in your repertory `./tests` and try the command `make test` and you will have a better error message that explains you what doesn't work, I suppose.

**Comment by Kagamihime** [comment]
You have to rebase your branch though

**Comment by DavidVn64** [comment]
Well this is odd because I see my branch David/tests-iotext up to date

**Comment by Kagamihime** [comment]
That's because you haven't updated your branch master

**Comment by DavidVn64** [comment]
Well I did, 2 minutes ago

**Comment by Kagamihime** [comment]
So you didn't do it correctly or didn't pushed your branch...

**Comment by Kagamihime** [comment]
I will do it

**Comment by Kagamihime** [comment]
Please, don't push anything on this branch ^^

**Pull request merged**